modules/client-configuration/pages/contact-methods-saltbundle.adoc

```diff
@@ -0,0 +1,81 @@
+[[contact-methods-salt]]
+= Salt Bundle
+
+== What is Salt Bundle?
+
+The Salt Bundle is a single binary package containing Salt Minion, Python 3, required Python modules and libraries.
+
+As the Salt Bundle is shipped with Python 3 and all the requirements for Salt to run, the Salt Bundle is not using the Python version coming from the system, so it can be installed on clients which doesn't meet the requirements for a given Salt version.
+
+It is also possible to install Salt Bundle on the system where Salt Minion is being used already to manage that client by the different Salt Master than the {productname} Server.
+
+== Client Registration with Salt Bundle as a Minion
+
+The registration method with the Salt bundle is the recommended registration method as long as it covers your use case.
+This section explains the advantages and limitations of the current implementation.
+The Salt bundle is provided as the [package]``venv-salt-minion`` that consists of Salt, Python, and the Python modules Salt depends on.
+Bootstrapping with {webui} is using Salt bundle as well, so bootstrapping with {webui} is not Python dependant.
+Using the Salt bundle, it is no longer needed that the client provides the required Python version or any Python modules.
+
+If you bootstrap new clients, registration with the Salt bundle is the default method.
+You can switch existing clients to the Salt bundle method.
+If you switch, the [package]``salt-minion`` package and its Python version and the Python modules will stay installed.
+
+=== Switching from Salt Minion to Salt Bundle
+
+The Salt state [literal]``util.mgr_switch_to_venv_minion`` is available to switch from [package]``salt-minion`` to [package]``venv-salt-minion``.
+It is recommended to switch to [package]``venv-salt-minion`` in two steps to avoid trouble with shifting processes:
+
+.Procedure: Switching with [literal]``util.mgr_switch_to_venv_minion`` state to [package]``venv-salt-minion``
+
+. Apply [literal]``util.mgr_switch_to_venv_minion`` with no pillar specified first.
+  This will result in the switch to [package]``venv-salt-minion`` with copying configuration files etc.
+  It will not clean up the original [package]``salt-minion`` configurations and its packages.
++
+----
+salt <minion_id> state.apply util.mgr_switch_to_venv_minion
+----
+. Apply [literal]``util.mgr_switch_to_venv_minion`` with [literal]``mgr_purge_non_venv_salt`` set to [literal]``True`` to remove [package]``salt-minion`` and with [literal]``mgr_purge_non_venv_salt_files`` set to [literal]``True`` to remove all the files related to [package]``salt-minion``.
+This second step repeats the first step, and then removes the old configuration files and the now obsolete [package]``salt-minion`` package.
```

Comment:
This can't be done in 1 step as it requires starting new minion service and stopping the old one. In general the second step is not repeating, but checking it the steps from first step were performed. Maybe need to rephrase it somehow.
```
The second step ensures that the first step passed and no changes required for it ....
```